src/core/bluetoothreceiver.cpp

```diff
@@ -0,0 +1,162 @@
+
+  emit lastGnssPositionInformationChanged( mLastGnssPositionInformation );
+}
+
+GnssPositionInformation BluetoothReceiver::fromQGeoPositionInfo( const QString &name )
+{
+
+  QGeoPositionInfoSource *positionSource = QGeoPositionInfoSource::createSource( name, nullptr );
+
+  QGeoPositionInfo info = positionSource->lastKnownPosition();
+  //not handled: magneticVariation = positionInfo.attribute( QGeoPositionInfo::Attribute::MagneticVariation );
+  //not handled: info.attribute( QGeoPositionInfo::Attribute::VerticalSpeed );
+  GnssPositionInformation gnssPositionInformation = GnssPositionInformation( info.coordinate().latitude(), info.coordinate().longitude(), info.coordinate().altitude(), info.attribute( QGeoPositionInfo::Attribute::GroundSpeed ),
+      info.attribute( QGeoPositionInfo::Attribute::Direction ), QList<QgsSatelliteInfo>(), 0, 0, 0,
+      info.attribute( QGeoPositionInfo::Attribute::HorizontalAccuracy ), info.attribute( QGeoPositionInfo::Attribute::VerticalAccuracy ),
+      info.timestamp(), QChar(), 0, -1, 0, QChar(), QList<int>(), true, info.attribute( QGeoPositionInfo::Attribute::VerticalSpeed ), info.attribute( QGeoPositionInfo::Attribute::MagneticVariation ), name );
+  qDebug() << "Qt Position DATA " << gnssPositionInformation.latitude() << "lon" << gnssPositionInformation.longitude() << "ele" << gnssPositionInformation.elevation();
+  return gnssPositionInformation;
+}
+
+#ifndef Q_OS_ANDROID
+void BluetoothReceiver::repairDevice( const QBluetoothAddress &address )
+{
+
+  connect( mLocalDevice.get(), &QBluetoothLocalDevice::pairingFinished, this, &BluetoothReceiver::pairingFinished, Qt::UniqueConnection );
+  connect( mLocalDevice.get(), &QBluetoothLocalDevice::pairingDisplayConfirmation, this, &BluetoothReceiver::confirmPairing, Qt::UniqueConnection );
+  connect( mLocalDevice.get(),  &QBluetoothLocalDevice::error, [ = ]( QBluetoothLocalDevice::Error error )
+  {
+    qDebug() << "RE-PAIR FAILED " << error;
+  } );
+
+  if ( mLocalDevice->pairingStatus( address ) == QBluetoothLocalDevice::Paired )
+  {
+    mLocalDevice->requestPairing( address, QBluetoothLocalDevice::Unpaired );
+  }
+  else
+  {
+    mLocalDevice->requestPairing( address, QBluetoothLocalDevice::Paired );
+  }
+}
+
+void BluetoothReceiver::confirmPairing( const QBluetoothAddress &address, QString pin )
+{
+  Q_UNUSED( address );
+  Q_UNUSED( pin );
+  mLocalDevice->pairingConfirmation( true );
+}
+
+void BluetoothReceiver::pairingFinished( const QBluetoothAddress &address, QBluetoothLocalDevice::Pairing status )
+{
+  if ( status == QBluetoothLocalDevice::Paired )
+  {
+    qDebug() << "paired device";
+    disconnect( mLocalDevice.get(), &QBluetoothLocalDevice::pairingFinished, this, &BluetoothReceiver::pairingFinished );
+    connectService( address );
+  }
+  else
+  {
+    qDebug() << "unpaired device";
```

Comment:
Can you check the debug statements and only leave where you think it will be useful in the future?

Reply:
Sorry, yes, this was intended. I just left them because it's easy to sort them out in the end.

Reply:
Btw. this current one I will keep. Since it's in a function used anyway only in the development environment (linux).